homeassistant/components/octoprint/button.py

```diff
@@ -124,5 +127,71 @@ async def async_press(self) -> None:
             await self.client.cancel_job()
 
 
+class OctoprintShutdownSystemButton(OctoprintButton):
+    """Shutdown the system."""
+
+    def __init__(
+        self,
+        coordinator: OctoprintDataUpdateCoordinator,
+        device_id: str,
+        client: OctoprintClient,
+    ) -> None:
+        """Initialize a new OctoPrint button."""
+        super().__init__(coordinator, "Shutdown System", device_id, client)
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        await self.client.shutdown()
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return self.coordinator.last_update_success
```

Comment:
I'm pretty sure this is the default behavior and is not needed? This goes for all of these

Reply:
No that's not default behavior. The default also checks if the printer is connected. But since this button is only for controlling OctoPrint and not the printer it would not make sense only being able to e.g. shutdown OctoPrint if it has a connection to the printer.